From e9d3532741b018cb746ed03f408b8e807aa20732 Mon Sep 17 00:00:00 2001
From: Andrew Liu <shengping.liu@windriver.com>
Date: Thu, 9 Sep 2010 15:16:06 +0800
Subject: [PATCH 192/252] pme: Reinsert pme error interrupt handler

Patch taken from FSL vendor SDK 2.1.

FMan code is now updated to handle shared interrupts.

Signed-off-by: Jeffrey Ladouceur <Jeffrey.Ladouceur@freescale.com>
Integrated-by: Andrew Liu <shengping.liu@windriver.com>
---
 drivers/match/pme2_ctrl.c |   10 +---------
 1 files changed, 1 insertions(+), 9 deletions(-)

diff --git a/drivers/match/pme2_ctrl.c b/drivers/match/pme2_ctrl.c
index 2d13c71..067a923 100644
--- a/drivers/match/pme2_ctrl.c
+++ b/drivers/match/pme2_ctrl.c
@@ -1,4 +1,4 @@
-/* Copyright (c) 2008, 2009, 2010, Freescale Semiconductor, Inc.
+/* Copyright (c) 2008-2010 Freescale Semiconductor, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -234,7 +234,6 @@ static struct of_device_id of_fsl_pme_ids[] = {
 MODULE_DEVICE_TABLE(of, of_fsl_pme_ids);
 
 /* Pme interrupt handler */
-#if 0
 static irqreturn_t pme_isr(int irq, void *ptr)
 {
 	static u32 last_isrstate;
@@ -254,7 +253,6 @@ static irqreturn_t pme_isr(int irq, void *ptr)
 	pme_out(global_pme, IER, ~last_isrstate);
 	return IRQ_HANDLED;
 }
-#endif
 
 static int of_fsl_pme_remove(struct of_device *ofdev)
 {
@@ -264,9 +262,7 @@ static int of_fsl_pme_remove(struct of_device *ofdev)
 	/* Disable PME..TODO need to wait till it's quiet */
 	pme_out(global_pme, FACONF, PME_FACONF_RESET);
 	/* Release interrupt */
-#if 0
 	free_irq(pme_err_irq, &ofdev->dev);
-#endif
 	/* Remove sysfs attribute */
 	pme2_remove_sysfs_dev_files(ofdev);
 	/* Unmap controller region */
@@ -325,13 +321,11 @@ static int __devinit of_fsl_pme_probe(struct of_device *ofdev,
 	pme_out(global_pme, SMCR, 0x00000211);
 
 	/* Register the pme ISR handler */
-#if 0
 	err = request_irq(pme_err_irq, pme_isr, IRQF_SHARED, "pme-err", dev);
 	if (err) {
 		dev_err(dev, "request_irq() failed\n");
 		goto out_unmap_ctrl_region;
 	}
-#endif
 
 #ifdef CONFIG_FSL_PME2_SRE_AIM
 	srec_aim = 1;
@@ -420,10 +414,8 @@ out_stop_accumulator:
 		cancel_delayed_work_sync(&accumulator_work);
 	}
 out_free_irq:
-#if 0
 	free_irq(pme_err_irq, &ofdev->dev);
 out_unmap_ctrl_region:
-#endif
 	pme_out(global_pme, FACONF, PME_FACONF_RESET);
 	iounmap(global_pme);
 	global_pme = NULL;
-- 
1.6.5.2

